src/hotspot/share/opto/escape.cpp

```diff
@@ -1279,7 +1301,24 @@ bool ConnectionGraph::reduce_phi_on_safepoints_helper(Node* ophi, Node* cast, No
   return true;
 }
 
-void ConnectionGraph::reduce_phi(PhiNode* ophi, GrowableArray<Node *>  &alloc_worklist, GrowableArray<Node *>  &memnode_worklist) {
+void ConnectionGraph::reduce_phi(PhiNode* ophi, GrowableArray<Node *>  &alloc_worklist, GrowableArray<Node *>  &memnode_worklist, Unique_Node_List &reducible_merges) {
+  Unique_Node_List nested_phis;
+  // Collect nested phi nodes first because the graph will change while splitting the child/nested phi node.
+  for (DUIterator_Fast imax, i = ophi->fast_outs(imax); i < imax; i++) {
+    Node* use = ophi->fast_out(i);
+    if (use->is_Phi()) {
+      assert(use->_idx != ophi->_idx, "Unexpected selfloop Phi.");
```

Comment:
Should we bailout of the reduction process if we somehow end up in this situation?
I.e., in a debug build we'll assert, but in a product build you're just ignoring the problem.